static/views/LockDeploy.jsx

```diff
@@ -0,0 +1,122 @@
+import {useCallback, useEffect, useMemo, useState} from 'react';
+import {useLocation, useNavigate, useOutletContext} from 'react-router-dom';
+
+import useApi from 'app/hooks/useApi';
+import useLocalStorage from 'app/hooks/useLocalStorage';
+
+function LockDeploy() {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const {appList, fetchApps} = useOutletContext();
+
+  const api = useApi();
+  const firstApp = appList.length !== 0 ? appList[0] : null;
+
+  const [lastApp, setLastApp] = useLocalStorage('lastApp', null);
+
+  const defaultApp = location.query?.app ?? lastApp ?? firstApp?.name ?? null;
+
+  const [app, setApp] = useState(defaultApp);
+
+  const appObj = useMemo(
+    () => appList.find(appEntry => appEntry.name === app),
+    [app, appList]
+  );
+
+  // XXX: boolean false is used to indicate
+  const [reason, setReason] = useState(appObj?.lockedReason);
+  const isLocked = !!appObj.lockedReason;
+
+  // update reason when app changes changes
+  useEffect(() => void setReason(appObj?.lockedReason ?? null), [appObj]);
```

Comment:
```suggestion
  useEffect(() => void setReason(appObj?.lockedReason ?? null), [setReason, appObj]);
```

Reply:
the lint rule doesn't complain since `setReason` is stable